### PR TITLE
Don't save config metadata for custom data sources

### PR DESCRIPTION
**Pull request by esoergel**
Re https://github.com/dimagi/commcare-hq/pull/6061
@czue

### CODE REVIEW

File: corehq/apps/userreports/tasks.py [hunk 1]
```diff
@@ -18,10 +18,11 @@ def rebuild_indicators(indicator_config_id):
...
 
-    # Save the start time now in case anything goes wrong. This way we'll be
-    # able to see if the rebuild started a long time ago without finishing.
-    config.meta.build.initiated = datetime.datetime.now()
-    config.save()
+    if not is_static:
+        # Save the start time now in case anything goes wrong. This way we'll be
+        # able to see if the rebuild started a long time ago without finishing.
+        config.meta.build.initiated = datetime.datetime.now()
+        config.save()
```
Comment (czue) [inline corehq/apps/userreports/tasks.py]: no big deal but this could just be moved to the else above.

File: corehq/apps/userreports/models.py [hunk 2]
```diff
@@ -331,7 +331,8 @@ def by_domain(cls, domain):
...
     def by_id(cls, config_id):
-        matching = [ds for ds in cls.all() if ds.get_id == config_id]
-        if not matching:
-            return None
-        return matching[0]
+        for ds in cls.all():
+            if ds.get_id == config_id:
+                return ds
+        raise BadSpecError(_('The data source referenced by this report could '
+                             'not be found.'))
```
Comment (czue) [inline corehq/apps/userreports/models.py]: this change makes sense - will wait on build unless you've run UCR tests

File: corehq/apps/userreports/tasks.py (hunk 1 above)
Comment (esoergel) [inline corehq/apps/userreports/tasks.py]: updated

File: corehq/apps/userreports/models.py (hunk 2 above)
Comment (esoergel) [inline corehq/apps/userreports/models.py]: They passed locally, but as long as it's in before deploy, I don't really care

**Comment by NoahCarnahan** [comment]
Thanks Ethan!

**Pull request merged**